wallet-new/src/Cardano/Wallet/API/V1/Errors.hs

```diff
@@ -73,6 +80,23 @@ data WalletError =
     -- node is still syncing with the blockchain.
     deriving (Show, Eq)
 
+convertTxError :: TxError.TxError -> WalletError
+convertTxError err = case err of
```

Comment:
This error conversion function seems like a bit of a smell to me.

Why not have a constructor in `WalletError` that takes a value of type `TxError.TxError` to save converting?

Reply:
I have a feeling that a single flat type might be easier to deal with than nested types. 

Really want an open variant though :(

Reply:
A single flat type makes sense when the errors are in a single package, but nesting errors like I suggest is necessary when dealing with errors outside the current package.

The problem with open variants is that there is no way for the compiler to check for pattern match completeness. The fact that sum types are closed is actually a nice property.